src/bp/ui-admin/src/Pages/Bot/Details.jsx

```diff
@@ -124,8 +137,27 @@ class Bots extends Component {
     )
   }
 
+  cancel = () => {
+    if (this.state.formHasChanged) {
+      // [TODO] max.cloutier 2020.01.17 Implement and use a custom confirm popup and replace the window.confirm in this file/app-wide
+      const conf = window.confirm(`There are unsaved changes in this form. Are you sure you want to cancel?`)
+
+      if (conf) {
+        this.backToList()
+      }
+    } else {
+      this.backToList()
+    }
```

Comment:
Changing the confirm dialog would be a good idea, the default one is ugly and we use it in a lot of places. You could avoid duplicating the method call with something like this:

```js
if(hasChanged && !window.confirm()){
  return
}
this.backToList()
```